lib/autoform.ex

```diff
@@ -17,38 +17,72 @@ defmodule Autoform do
   """
   defmacro __using__(_opts) do
     quote location: :keep do
+      import Ecto.Query, only: [from: 2]
+
       @doc """
-        Renders a 'new' or 'update' form based on the schema passed to it.
+        Renders a 'create' or 'update' form based on the schema passed to it.
+
+        ## Examples
 
-        Example:
           In a controller:
-            render_autoform(conn, :update, User, user: get_user_from_db())
+
+            render_autoform(conn, :update, User, assigns: [user: get_user_from_db()])
 
           In a template:
-            <%= render_autoform(@conn, :new, User, changeset: @changeset) %>
-      """
-      @spec render_autoform(Plug.Conn.t(), atom, Ecto.Schema.t(), Keyword.t() | map) ::
-              Plug.Conn.t()
-      def render_autoform(conn, action, schema, assigns) when action in [:update, :create] do
-        fields =
-          schema.__schema__(:fields) |> Enum.reject(&(&1 in [:id, :inserted_at, :updated_at]))
 
-        {_key, schema_data} =
+            <%= render_autoform(@conn, :create, User, assigns: [changeset: @changeset)], exclude: :date_of_birth %>
+
+        ## Options
+
+          * `:assigns` - The assigns for the form template, for example a changeset for update forms. Will default to empty list
+          * `:exclude` - A list of any fields in your schema you don't want to display on the form
+          * `:update_field` - The field from your schema you want to use in your update path (/users/some-id), defaults to id
+          * `:assoc_query` - An ecto query you want to use when loading your associations
```

Comment:
how do `:assoc_query`s get "sanitised"? 
or is the fact that you can only run them through `autoform` by writing compiled Elixir code "enough" for security?